src/Rewrite.hs

```diff
@@ -51,12 +51,7 @@ version log (Args env attrPth drvFile drvContents) = do
       lift $ log "[version] generic version rewriter does not support multiple hashes"
       return Nothing
     else do
-      oldHash <- Nix.getOldHash attrPth
-      _ <- lift $ File.replace (Utils.oldVersion env) (Utils.newVersion env) drvFile
-      _ <- lift $ File.replace oldHash Nix.sha256Zero drvFile
-      newHash <- Nix.getHashFromBuild attrPth
-      when (oldHash == newHash) $ throwE "Hashes equal; no update necessary"
-      _ <- lift $ File.replace Nix.sha256Zero newHash drvFile
+      srcVersionFix (Args env attrPth drvFile drvContents)
```

Comment:
Instead of repacking args we can give args an name when we are destructuring it with `args@(Args env attrPth drvFile drvContents)`.